src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Shared/Authorization/UserInfo.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+
+namespace BlazorWasm_CSharp.Shared.Authorization
+{
+    public class UserInfo
+    {
+        public static readonly UserInfo Anonymous = new UserInfo();
+
+        public UserInfo() { }
+
+        public UserInfo(ClaimsPrincipal claimsPrincipal)
```

Comment:
What do you think about maybe moving this "populate from claimsPrincipal" logic into the server project (in `UserController`'s `GetCurrentUser`)? This would just be to clarify that the server is the only thing that controls what data gets copied in here - the client doesn't control it, it just receives some serialized data.

As a more minor thing, this logic just isn't needed on the client and won't be stripped out by the linker since the linker doesn't run on user assemblies.

Reply:
I've moved this. Not sure why the diff is wrong.